### PR TITLE
fix(ci): make the terminology and cron-prompt guards fail on what they check, and pass the version arg to docker targets

### PR DESCRIPTION
**Pull request by dshnayder**
## Summary

Two CI guards now fail on what they check instead of passing on what they could not read, the prompt-assets check learns the `deliver` field, and the docker `make` targets pass the version argument every other image entry point already passes.

## Why This Change

`hack/check-docs-terminology.sh` treated a grep that could not run the same as a grep that found nothing, and anchored its cron-prompt quotation check on a phrase inside the prompt, so a rewording of the prompts would have made the loop run zero times and go green. `scripts/check_prompt_assets.py` skipped a cron path with an argument after it or an interpreter before it, and did not look at `deliver` at all, though a misspelled part next to a resolvable one is dropped by the scheduler with the run still recording `ok`. `make docker-build` shipped images that report `dev` as their version on the wire.

## What Changed

- The terminology guard records grep failures and reports them once, finds fenced roster quotations structurally through the new `hack/scan-cron-prompts.awk` (keyed on the JSON `"id"` and the keys only a roster entry has, `schedule`, `skills`, `deliver`, `no_agent`, `risk`, with fences matched by character and length as CommonMark closes them), refuses a fenced entry naming no known job (a placeholder id in angle brackets, `"<your-audit>"`, marks an illustration and is left ungraded), and refuses an elision under `MIN_QUOTED_CHARS`. It accepts extra documents through `DOCS_TERMINOLOGY_EXTRA_FILES`, appended to the tracked set, which is how its verdicts are tested against the real rosters. Tests drive the awk against fixtures across gawk, mawk and awk, and drive the script against a document built to fail each check.
- `check_prompt_assets.py` gains `check_cron_delivery` (`CRON_DELIVER_VALUES`, platform names and `all` case-folded, `local`/`origin` as spelled, matching the pinned scheduler), a cron path extractor that finds a path between whitespace or quotes wherever it stands in the prompt (after an interpreter, before an argument, after an apostrophe in the prose), and named roster-shape errors.
- The Makefile's docker targets pass `KUBE_AGENTS_VERSION` (default `dev`); a dry-run test pins it beside the two Cloud Build configs.
- `agents/platform/cron/README.md` names the new gate; the docs map's description of the terminology check names the awk.

## Context

Extracted from Adam Parco's `adamparco/kube-agents:fleet-audit-collectors` (7a53ac28) as one of the independent slices of that branch, with his agreement. Rebased onto `main` past #1574 and #1663 (the latter added constants beside this branch's in `tests/test_remote_mcp_user_agent.py`; both kept). #1627, #1304 and #1368 touch other regions of `check_prompt_assets.py`; merge-conflict risk only.

## Testing

- `tests/test_docs_terminology_guard.py` (29), `scripts/test_check_prompt_assets.py` (80), `tests/test_remote_mcp_user_agent.py` (5), `scripts/test_sync_upstream_skills.py` pass; `hack/check-docs-terminology.sh`, `scripts/check_prompt_assets.py`, `make prompt-check`, `make shellcheck`, `scripts/generate_docs.py --check` and `make docs-check` run clean.
- Each new verdict test was run with its fix reverted and went red: `search()` folding stderr away, the value extraction reading only the last value on a line, the length floor at zero, the verbatim check disabled, the quote-pairing extractor restored, an unknown id refusing on its own, any marker closing a fence, and the unterminated-value fallback deleted.
- The full `make test-python` sweep on a merge of this branch with its siblings fails the same twelve environment-bound tests as `main` and no others.

### Live validation

Not live-tested: the change is CI guards, their tests, and a build argument, none of which reach a running installation. The guards were run against the tree as CI runs them.

## Self-Review

Two rounds of `review-adversarial` and `review-docs-drift`, each in a fresh subagent handed only the diff range and the docs-check result.

- **A generated-footer edit in `sync-upstream-skills.py` redded `test_sync_upstream_skills`** (CONFIRMED, BLOCKER): hunk dropped.
- **Docstrings and the guard's banner described a roster and a test name that exist only on the source branch** (CONFIRMED, HIGH): rewritten against `main`.
- **The `cron-deliver` rationale misstated the scheduler**: a wholly unresolvable `deliver` is a recorded error the delivery watch grades; the silent shape is a bad part beside a good one (CONFIRMED, HIGH): docstrings, message, README and test docstring corrected. Case handling verified against the pinned scheduler read out of the built image.
- **A dead `fleet-audit-reports` entry in `generate_docs.py`, out-of-scope ignore-file hunks, a duplicate audit-delivery test, and a real-tree test that made the Python suite a second reporter of every terminology failure**: all dropped or softened.
- **From the bot's first pass** (three Medium, all fixed): the quote-pairing extractor lost every path after an unpaired quote earlier in the prompt, so quotes are now boundaries rather than brackets, with an apostrophe test and a bare-path test; the grep-failure path and the stale, short and two-values-per-line grading had no executing test, so the guard gained the extra-files seam and six verdict tests that run it against the real rosters with one injected document each. Its low items: the wiring test now matches the awk invocation rather than the existence check; the comment claiming the `"id"` sits one line above the `"prompt"` says "in the same block"; `"prompt": ""`, what a `no_agent` entry renders, is no longer an elision; the make targets have the dry-run test; `check_cron_delivery` runs in the fixture's `run()`; the `_audit_deliveries` helper and its self-test, which grade a rule that lands with the fleet-audit chain, are dropped. The docker `--build-arg` hunk stays: it is the third change the title names, and it travelled with the guards on the source branch.
- **From the bot's second pass** (low, all fixed): the dry-run `make` test names `REPO`, so the sweep no longer evaluates the `gcloud` shell default; the verdict tests' negative assertions are scoped to the fixture document, so a stale quotation elsewhere in the tree is the guard's finding and not this suite's; the ordering test anchors on the roster read, the first exit it guards; the roster-shape error says the run stops; a comment no longer cites a line number.
- **From the bot's third pass** (two Medium, fixed): a fenced entry trimmed to `name`/`risk`/`enabled` with its `id` line dropped set only the request-body flag and left the scan ungraded and unreported, so every roster key now marks an entry, with a fixture test; `check_cron_delivery` was in `main()`'s sum on a line no test reached, so a test now runs a roster carrying `deliver: "gchat"` through `main()` and reads the `cron-deliver` finding, and goes red with the line deleted.
- **From the bot's fourth pass** (three Medium, fixed): a known id anywhere in a fence carried the whole block to `R`, so a renamed entry excerpted beside a current one had its prompt graded and its stale id unreported; an unknown id now refuses the block, with a two-entry fixture test. The orphan refusal was tested in the awk only, so a verdict test now hands the script an entry naming a job the roster lacks and reads the error. The short-elision test's negative assertion was tree-wide; every verdict assertion now reads the hit lines listed under its own `::error::` and looks for the fixture there, on both sides.
- **From the human review** (jayantid): a contributor rendering an example entry got the orphan error with advice that fits only a stale quotation, and had no way to show an entry without naming a job. A placeholder id in angle brackets now marks an illustration the scan leaves ungraded, the error names that spelling, the docs map states it, and a fixture test pins it. The suggestion to generate the three hand-pasted roster examples from `jobs.json` and retire the cron-prompt section was withdrawn from this change as a follow-up.
- **From the bot's fifth pass** (Medium, fixed): a bare path touching sentence punctuation (`governance/x.md.`, a comma, a closing parenthesis) failed the extractor's boundary and went unchecked; clause punctuation is now a boundary on either side when a boundary of its own follows it, so `x/y.md.bak` stays one token. Tested for four shapes and the suffix case.
- **From the bot's sixth pass**: a placeholder id anywhere in a fence switched grading off for a real or renamed entry beside it (Medium, fixed): the placeholder now ranks below a real and an unknown id, so such a block is graded or refused as if the placeholder were not there, with a two-entry test. Its High, that GNU grep 3.8+ would write a stray-backslash warning for the finding-id probe `'\[a-z0-9\._-\]'` and `search()` would read it as a failed check, does not reproduce: GNU grep 3.12 writes nothing to stderr for that probe through the `xargs -0 grep` shape `search()` uses, and the Documentation Checks and Python Unit Tests jobs, which run the guard and its verdict tests on `ubuntu-latest`, passed on this head. Left as spelled.
- **From the bot's seventh pass** (three Medium, fixed): marking a block as a roster entry on `name` or `enabled`, or on an unknown `"id"` alone, would refuse a bench scenario, a tool definition or a payload rendered in a fence, against what the wrapper's comment promised; the signal is now the roster-only keys, an unknown id counts only beside one, both comments state the rule and its residual hole, and two tests render the unrelated shapes. Fence tracking toggled on every marker line, so a four-backtick block showing a single ``` line inverted the scan for the rest of the file; fences now close only on their own character at their own length or longer, with two tests. The unterminated-value fallback had no test; a verdict test feeds a value with no closing quote and reads it under the verbatim error.
- **Reported, not changed**: hard-coded counts in the banner were removed where they were already off; two comment citations of a not-yet-landed adapter function were reworded.

Generated with the help of Claude Fable 5.1 (Claude Code), directed by Dmitry Shnayder; the change itself originates from Adam Parco's branch.

## Risk & Rollout

Affects pull-request checks only. The stricter guard can red a pull request that renders an illustrative roster entry with a placeholder id; the failure message says how to fix it. No runtime code path is touched.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: scripts/check_prompt_assets.py [hunk 1]
```diff
@@ -199,9 +204,55 @@
...
+# reference would leave `make prompt-check` at exit 0.
+CRON_PROMPT_SPAN = re.compile(r"['\"`]([^'\"`\n]+)['\"`]")
+# Whitespace or the span edge on both sides, so a token is a path only when it
+# stands alone. That also drops the absolute interpreter above rather than
+# reporting `/opt/hermes/...` as an asset missing from the profile: a leading
+# `/` fails the first segment, and there is no whitespace inside it to restart
+# the match on a suffix.
+CRON_PROMPT_PATH = re.compile(
+    r"(?:\A|(?<=\s))([A-Za-z0-9_.-]+(?:/[A-Za-z0-9_.-]+)+\.(?:md|json|py|sh))(?=\Z|\s)"
+)
```
[the remark this answers is not in the document]
Comment (dshnayder) [reply scripts/check_prompt_assets.py]: Confirmed and fixed in e2a5e3f2. The span stage is gone: `CRON_PROMPT_PATH` now takes whitespace or a quote as its boundary on either side and is applied to the whole prompt, so an unpaired quote earlier in the prose cannot consume a path's opening quote. Two tests added, one with `Don't` before the quoted path and one with a bare path in the prose; with the two-stage extractor restored the apostrophe test goes red. Both rosters still pass `make prompt-check` with no new finding.

File: hack/check-docs-terminology.sh [hunk 2]
```diff
@@ -35,9 +43,26 @@ fi
...
+  local out
+  : > "$GREP_ERR"
+  out=$(tr '\n' '\0' < "$FILE_LIST" | xargs -0 grep -nEI -H "$1" 2>"$GREP_ERR")
+  if [ -s "$GREP_ERR" ]; then
+    printf 'pattern: %s\n' "$1" >> "$GREP_FAILURES"
+    sed 's/^/  /' "$GREP_ERR" >> "$GREP_FAILURES"
+    return 2
+  fi
+  printf '%s' "$out"
 }
```
[the remark this answers is not in the document]
Comment (dshnayder) [reply hack/check-docs-terminology.sh:79]: Fixed in e2a5e3f2. The guard accepts extra documents through `DOCS_TERMINOLOGY_EXTRA_FILES` (appended to the tracked set, so the copy floors still see the real documents), and `test_a_file_grep_cannot_read_is_a_check_that_did_not_run` hands it a path that does not exist: exit 1, the `could not run` report naming the file, and no `No document states this cap` line from the gated floors. With `search()` reverted to `2>/dev/null || true` the test goes red. The real-tree test now also refuses `could not run`, and the wiring test matches the awk invocation rather than the existence check.

File: hack/check-docs-terminology.sh [hunk 3]
```diff
@@ -357,21 +411,150 @@ for JOBS_FILE in $CRON_JOBS; do
...
+    MATCH_STATUS=$?
+    if [ "$MATCH_STATUS" -ge 2 ]; then
+      echo "ERROR: grep could not read ${CRON_JOBS}; the cron-prompt guard cannot run." >&2
+      exit 1
+    fi
+    if [ "$MATCH_STATUS" -ne 0 ]; then
+      STALE_PROMPTS="${STALE_PROMPTS}${HIT}
+"
+      break
+    fi
```
[the remark this answers is not in the document]
Comment (dshnayder) [reply hack/check-docs-terminology.sh]: Fixed in e2a5e3f2 through the same seam: five verdict tests run the script against the real rosters with one injected document each, a verbatim control, a drifted value, an `R…` elision, a `no_agent` entry's `"prompt": ""` (now exempt from the floor, as you noted), and a line carrying a fabricated first value beside a verbatim second one. Each assertion is on the error the check prints, not the exit status alone. Reverting the value extraction to the last value only, the floor to zero, or the verbatim check to always-match turns the matching test red.

File: hack/scan-cron-prompts.awk [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    seg = substr(rest, s, l)
+    sub(/^"id"[ \t]*:[ \t]*/, "", seg)
+    # An id the roster does not know is a renamed or mistyped job, which is
+    # exactly what `O` is for -- and it is a roster entry whether or not a
+    # sibling key survived the trim.
+    if (seg in ids) { hasid = 1; dochasid = 1 } else cronish = 1
+    rest = substr(rest, s + l)
+  }
+  if ($0 ~ /"(schedule|skills|deliver|no_agent)"[ \t]*:/) cronish = 1
+  # Any JSON key on the line other than `prompt`, which switches off the
```
[the remark this answers is not in the document]
Comment (dshnayder) [reply hack/scan-cron-prompts.awk:169]: Confirmed and fixed in 0c4cbf29: the scan now treats every roster key as marking an entry, `name`, `risk` and `enabled` beside the four routing keys, so a copy trimmed to those three with the `id` line dropped is reported as an orphan. Fixture test added (`test_an_entry_trimmed_to_name_risk_and_enabled_is_still_reported`), red with the old key list; the tree still passes the guard with no new orphan.

File: scripts/check_prompt_assets.py [hunk 5]
```diff
@@ -788,6 +961,7 @@ def main(argv: list[str] | None = None) -> int:
         + check_skill_refs(files, skills)
         + check_skill_manifests(skills)
         + check_cron_assets()
+        + check_cron_delivery()
         + check_cron_risk()
```
[the remark this answers is not in the document]
Comment (dshnayder) [reply scripts/check_prompt_assets.py:971]: Fixed in 0c4cbf29. `test_a_bad_deliver_part_fails_the_run_through_main` runs a fixture roster carrying `deliver: "gchat"` through `cpa.main([])` and reads the `cron-deliver` finding from stderr; with the `+ check_cron_delivery()` line deleted it goes red.

File: tests/test_docs_terminology_guard.py [hunk 6]
```diff
@@ -0,0 +1,485 @@
...
+                ```json
+                {{ "id": "{KNOWN_ID}", "prompt": "R…" }}
+                ```
+                """
+            ]
+        )
+        self.assertEqual(proc.returncode, 1, proc.stdout + proc.stderr)
+        self.assertIn(self.SHORT, proc.stdout)
+        self.assertNotIn(self.STALE, proc.stdout)
+
```
[the remark this answers is not in the document]
Comment (dshnayder) [reply tests/test_docs_terminology_guard.py:509]: Fixed in be323c27: every verdict assertion now reads the hit lines the guard lists under its own `::error::` (`listed_under`) and looks for the fixture there, on both sides -- the short-elision test asserts `doc0.md` is listed under the length floor and not under the verbatim check, whatever the rest of the tree says.

File: hack/check-docs-terminology.sh [hunk 7]
```diff
@@ -380,6 +583,12 @@ if [ -n "$STALE_PROMPTS" ]; then
   FAILED=1
 fi
 
+if [ -n "$SHORT_PROMPTS" ]; then
```
[the remark this answers is not in the document]
Comment (dshnayder) [reply hack/check-docs-terminology.sh:671]: Fixed in be323c27: `test_an_entry_naming_a_job_the_roster_lacks_is_refused_by_the_guard` hands the script an entry with an unknown id through the seam and reads the `names no job id` error with the fixture listed under it; with the orphan block disabled it goes red. `ORPHAN` is now read.

File: hack/scan-cron-prompts.awk [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    # runs, and a scan that silently stopped after the first id on a line would
+    # grade a one-line entry by nothing.
+    s = RSTART; l = RLENGTH
+    seg = substr(rest, s, l)
+    sub(/^"id"[ \t]*:[ \t]*/, "", seg)
+    # An id the roster does not know is a renamed or mistyped job, which is
+    # exactly what `O` is for -- and it is a roster entry whether or not a
+    # sibling key survived the trim.
+    if (seg in ids) { hasid = 1; dochasid = 1 } else cronish = 1
+    rest = substr(rest, s + l)
```
[the remark this answers is not in the document]
Comment (dshnayder) [reply hack/scan-cron-prompts.awk]: Confirmed and fixed in be323c27: an unknown id anywhere in a fence now refuses the whole block, known ids beside it or not (`badid` is tracked apart from `hasid`). The two entries are one excerpt and one edit fixes it, so both lines are reported. Fixture test added for a current and a renamed entry in one fence; with the old classification restored it goes red.

File: hack/check-docs-terminology.sh [hunk 9]
```diff
@@ -357,21 +424,157 @@ for JOBS_FILE in $CRON_JOBS; do
...
+fi
+
 PROMPT_HITS=$(mktemp)
-trap 'rm -f "$FILE_LIST" "$PROMPT_HITS"' EXIT
-search 'all [0-9]+ lines of it' > "$PROMPT_HITS"
+ORPHAN_HITS=$(mktemp)
+ROSTER_ID_FILE=$(mktemp)
+trap 'rm -f "$FILE_LIST" "$GREP_ERR" "$GREP_FAILURES" "$PROMPT_HITS" "$ORPHAN_HITS" "$ROSTER_ID_FILE"' EXIT
+
+# Which documents are claiming to quote the roster at all. The anchor below is
```
Comment (jayantid) [inline hack/check-docs-terminology.sh]: 🟡 **Technical design.** This hardens a guard over three hand-copied roster entries that the repository's own rule says should be generated, so the source of the problem stays.

The cron-prompt section exists because `concepts/autonomous-watchdogs.md:96`, `concepts/skills.md:53` and `reference/cron-jobs.md:55` each render the `compliance-audit` entry by hand. The third sits eight lines below a `BEGIN GENERATED: cron-jobs` region that `scripts/generate_docs.py` already fills from the same two rosters. AGENTS.md's documentation rule is not to hand-write what mirrors a machine-readable file. This change adds the awk scanner, the orphan rule, the length floor and their tests to police those three copies.

If it merges, the copies stay hand-made, every future quotation inherits the same guard, and the guard grows with each edge case the way this pass already did through four bot rounds.

Fix: keep this change, and open a follow-up that emits the fence from a `cron-job-example` generated region (the table code already truncates prompts, so the elided form in `skills.md` is one option away). The cron-prompt section of the guard then has nothing left to check.

File: hack/check-docs-terminology.sh [hunk 10]
```diff
@@ -357,21 +424,157 @@ for JOBS_FILE in $CRON_JOBS; do
...
+if ! SCAN=$(tr '\n' '\0' < "$FILE_LIST" \
+  | xargs -0 awk -v idfile="$ROSTER_ID_FILE" -f "$SCAN_AWK"); then
+  echo "ERROR: could not scan the documentation for cron prompts; the guard cannot run." >&2
+  exit 1
+fi
+printf '%s\n' "$SCAN" | sed -n 's/^R://p' > "$PROMPT_HITS"
+printf '%s\n' "$SCAN" | sed -n 's/^O://p' > "$ORPHAN_HITS"
+
+if [ -s "$ORPHAN_HITS" ]; then
+  echo "::error::A rendered cron roster entry names no job id from ${CRON_JOBS}, so its prompt cannot be checked. Restore the \"id\" line, or correct it if the job was renamed."
```
Comment (jayantid) [inline hack/check-docs-terminology.sh]: 🟡 **UX flow.** The new orphan refusal has no way to render an illustrative roster entry, and its message assumes the author is quoting a stale job.

The `O` rule refuses any fenced entry whose `"id"` is not on a roster. The Risk section says an illustrative entry with a placeholder id will red and "the failure message says how to fix it". The message says "Restore the `"id"` line, or correct it if the job was renamed", which describes a drifted quotation. It gives a contributor writing the how-to at `autonomous-watchdogs.md:141` no sanctioned way to show `"id": "<your-audit>"`. Sibling guards carry an escape: `AUDIENCE_EXEMPT_SITE_PAGES` in `check_docs_map.py`, `docs_audience_denylist.txt` for the audience check.

If it merges, the first how-to that renders an example entry reds CI with advice that does not apply, and the author either quotes a real job that is not the one they mean or drops the example.

Fix: one of two lines. Either the error names the convention ("quote a real entry, or render the example outside a fence"), or the awk treats an angle-bracket id like `"<job-id>"` as an example and skips the block.

File: hack/check-docs-terminology.sh (hunk 9 above)
Comment (jayantid) [inline hack/check-docs-terminology.sh:528]: 🟡 **Technical design.** This guard protects three hand-pasted copies of a roster entry that should be generated instead.

`concepts/autonomous-watchdogs.md:96`, `concepts/skills.md:53` and `reference/cron-jobs.md:55` each paste the `compliance-audit` entry from `jobs.json` by hand. The cron-prompt section of this guard exists to catch those copies going stale. `scripts/generate_docs.py` already reads both rosters and fills a generated table in `cron-jobs.md`, eight lines above one of the copies.

If the generator emitted the example fence too, there would be no hand copies and nothing here to check.

Fix: keep this change. Open a follow-up that adds a `cron-job-example` generated region and drops the hand copies. The cron-prompt section can go with them.

File: hack/check-docs-terminology.sh (hunk 10 above)
Comment (jayantid) [inline hack/check-docs-terminology.sh]: 🟡 **UX flow.** A contributor who renders an example roster entry gets a CI failure with advice that does not fit.

The new orphan rule refuses any fenced entry whose `"id"` is not on a roster. The error says "restore the id line, or correct it if the job was renamed". That fits a stale quotation. It does not fit someone writing the how-to at `autonomous-watchdogs.md:141` with `"id": "<your-audit>"` as an example, and there is no other way to show one. Sibling guards have an escape hatch, such as `AUDIENCE_EXEMPT_SITE_PAGES` in `check_docs_map.py`.

Fix: either add one sentence to the error saying how to render an example (quote a real job, or keep the example out of a fence), or have the awk skip an id in angle brackets like `"<job-id>"`.

File: hack/check-docs-terminology.sh (hunk 9 above)
Comment (jayantid) [reply hack/check-docs-terminology.sh:528]: Withdrawing this as a request on this PR. The generated fence can replace the hand copies later at the same cost, and this change did not create them, so it is a follow-up rather than something to change here. Leaving the note as a pointer for whoever picks up that follow-up. Resolving.

File: hack/check-docs-terminology.sh (hunk 10 above)
Comment (dshnayder) [reply hack/check-docs-terminology.sh]: Agreed, and done in 8780d85d the second way: an id in angle brackets (`"id": "<your-audit>"`) marks an illustration, and the scan leaves that block ungraded and unreported whatever else it carries. The orphan error now ends by naming that spelling, the docs map states the escape beside the rule, and a fixture test pins it (red with the escape removed). Thanks, this is the case the rule's wording had not considered.

File: hack/check-docs-terminology.sh (hunk 9 above)
Comment (dshnayder) [reply hack/check-docs-terminology.sh:528]: Thanks. Agreed it belongs in a follow-up: a `cron-job-example` generated region from `jobs.json` would retire the three hand copies and the cron-prompt section with them. Leaving your note as the pointer.

**Review by jayantid — APPROVED** [review]

File: hack/check-docs-terminology.sh (hunk 9 above)
Comment (dshnayder) [reply hack/check-docs-terminology.sh:528]: Filed as #1659.

File: scripts/check_prompt_assets.py [hunk 11]
```diff
@@ -194,15 +199,66 @@
...
+#
+# A leading `/` fails the first segment, so an absolute interpreter is not
+# itself an asset -- `/opt/hermes/.venv/bin/python3` ships in the image, not
+# the profile -- while the script after it is still found, because the space
+# between them is a boundary.
+CRON_PROMPT_PATH = re.compile(
+    r"(?:\A|(?<=[\s'\"`]))"
+    r"([A-Za-z0-9_.-]+(?:/[A-Za-z0-9_.-]+)+\.(?:md|json|py|sh))"
+    r"(?=\Z|[\s'\"`])"
 )
```
[the remark this answers is not in the document]
Comment (dshnayder) [reply scripts/check_prompt_assets.py:237]: Confirmed and fixed in 0026ac6e: clause punctuation is a boundary on either side of the path when a boundary of its own follows it, so `Read governance/x.md.`, a comma, a question mark and parentheses are all matched, while `x/y.md.bak` stays one token. `test_a_bare_path_ending_a_clause_is_still_checked` covers the four shapes and the suffix case, and goes red with the old lookahead.

File: hack/scan-cron-prompts.awk [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+# excerpt and one edit fixes it, so refusing both lines is the honest report.
+#
+# A placeholder id in angle brackets -- `"id": "<your-audit>"` -- marks an
+# illustration, not a quotation: a how-to has no other way to show the shape of
+# an entry without naming a job. Such a block is left ungraded and unreported,
+# whatever else it carries. The error the guard prints for an orphan names this
+# spelling, so a contributor who meant an example is told how to write one.
+function flush(   i, kind) {
+  kind = placeholder ? "" : (badid ? "O" : (hasid ? "R" : (cronish ? "O" : (otherkey ? "" : "?"))))
+  for (i = 1; i <= n; i++) {
```
[the remark this answers is not in the document]
Comment (dshnayder) [reply hack/scan-cron-prompts.awk:89]: Confirmed and fixed in 5eb94383: the placeholder now ranks below a real and an unknown id in `flush()`, so a block that carries a placeholder beside a quotation is graded on the real id, and refused on an unknown one, as if the placeholder were not there. Only a block whose ids are all placeholders is left ungraded. `test_a_placeholder_does_not_switch_off_a_real_entry_in_the_same_fence` covers both pairings and goes red with the old order.

File: hack/check-docs-terminology.sh [hunk 13]
```diff
@@ -223,19 +274,27 @@ cap_guard() {
...
-  copies=$(printf '%s' "$all" | grep -cE "$good" || true)
-  if [ "${copies:-0}" -lt 1 ]; then
-    echo "::error::No document states this cap any more; restore it or drop the guard. ($why)"
-    FAILED=1
+  # A guard that passes because every copy disappeared is not a passing guard --
+  # unless the search itself is what disappeared them. A broken pattern or an
+  # unreadable file makes `search` return nothing for every probe, and every
+  # cap guard below then announces that its cap is undocumented. Those
+  # confident wrong diagnoses bury the one line that names the real cause, which
+  # `search` has already recorded and the report at the end of this section
```
[the remark this answers is not in the document]
Comment (dshnayder) [reply hack/check-docs-terminology.sh:365]: This does not reproduce, so the probe stays as spelled. On GNU grep 3.12, `printf '%s\n' /tmp/probe.txt | tr '\n' '\0' | xargs -0 grep -nEI -H '\[a-z0-9\._-\]' 2>/tmp/err` (the exact shape `search()` uses) matches the line and leaves `/tmp/err` at zero bytes; the same probe run directly prints no warning. And the two jobs that would go red if the warning were emitted, Documentation Checks (which runs `make docs-check-terminology` on `ubuntu-latest`) and Run Python Unit Tests (which runs the verdict tests asserting the `could not run` block is absent), both passed on this pull request's head 0026ac6e. The lexer's default arm warns for a backslash before an ordinary letter; `]` is not treated as one there.

File: Makefile [hunk 14]
```diff
@@ -19,6 +19,16 @@ BASE_IMAGE_ARGS := $(foreach v,$(BASE_IMAGE_VARS),$(if $($(v)),--build-arg $(v)=
 SANDBOX_IMAGE_VARS := PYTHON_IMAGE
```
[the remark this answers is not in the document]
Comment (dshnayder) [reply Makefile:19]: Fixed in 5536eb10: `test_an_unterminated_value_is_still_graded` hands the guard a value with no closing quote and reads the fixture under the verbatim error; with the fallback deleted it goes red. Added to the reverted-fix list in Testing.

File: hack/scan-cron-prompts.awk [hunk 15]
```diff
@@ -0,0 +1,157 @@
...
+    else if (seg in ids) { hasid = 1; dochasid = 1 }
+    else badid = 1
+    rest = substr(rest, s + l)
+  }
+  # The roster schema, not only its routing keys. An entry trimmed to
+  # `name`/`risk`/`enabled` with the `id` line dropped used to set `otherkey`
+  # alone and fall out of the scan ungraded and unreported -- the shape `O`
+  # exists for.
+  if ($0 ~ /"(schedule|skills|deliver|no_agent|name|risk|enabled)"[ \t]*:/) cronish = 1
+  # Any JSON key on the line other than `prompt`, which switches off the
```
[the remark this answers is not in the document]
Comment (dshnayder) [reply hack/scan-cron-prompts.awk]: Confirmed and fixed in 5536eb10. The signal is now the keys only a roster entry has (`schedule`, `skills`, `deliver`, `no_agent`, `risk`); `name` and `enabled` no longer count, and an unknown `"id"` counts as a renamed job only beside one of those, so a bench scenario, a tool definition or a payload with its own id is left alone. The awk header and the wrapper's comment now state that rule and its residual hole (a quotation trimmed to `id` and `prompt` whose id was renamed). `test_an_unrelated_object_with_an_id_or_a_name_is_left_alone` renders both shapes and goes red with the old rule; the orphan fixtures carry a `schedule` so they still say what they said.

File: hack/scan-cron-prompts.awk [hunk 16]
```diff
@@ -0,0 +1,157 @@
...
+  while ((getline id < idfile) > 0) if (id != "") ids["\"" id "\""] = 1
+}
+
+FNR == 1 { endfile() }
+
+# `>` so a fence inside a blockquote still opens and closes a block. Without it
+# two blockquoted manifests were one block, and the id in the first graded the
+# prompt in the second.
+/^[ \t>]*(```|~~~)/ { flush(); infence = !infence; next }
+
```
[the remark this answers is not in the document]
Comment (dshnayder) [reply hack/scan-cron-prompts.awk:144]: Confirmed and fixed in 5536eb10: a fence now closes only on the character that opened it, at the opener's length or longer. `FenceLengthTest` covers a four-backtick block showing a single ``` line followed by a real quotation (graded, at its own line) and a tilde block holding backtick markers; both go red with the toggle restored. Verified across awk, mawk and gawk.
